app/views/handlers/primary_person_list_collector.py

```diff
@@ -26,18 +26,21 @@ def get_context(self):
         return build_question_context(self.rendered_block, self.form)
 
     def handle_post(self):
+        list_name = self.rendered_block['for_list']
+
         if (
             self.form.data[self.rendered_block['add_or_edit_answer']['id']]
             == self.rendered_block['add_or_edit_answer']['value']
         ):
             self._is_adding = True
             self.questionnaire_store_updater.update_answers(self.form)
             self._primary_person_id = self.questionnaire_store_updater.add_primary_person(
-                self.rendered_block['for_list']
+                list_name
             )
-            self.questionnaire_store_updater.save()
+
         else:
-            self.questionnaire_store_updater.remove_primary_person(
-                self.rendered_block['for_list']
-            )
-            return super().handle_post()
+            self.questionnaire_store_updater.remove_primary_person(list_name)
+            super().handle_post()
+
+        self.evaluate_and_update_section_status_on_list_change(list_name)
+        self.questionnaire_store_updater.save()
```

Comment:
This does mean when they are not adding we are potentially saving the state twice. Is it better to accept some duplication and avoid this?

Reply:
I think at this point it's best to check that is_dirty is True.